wasm/src/lib.rs

```diff
@@ -197,18 +197,16 @@ where
     }
 }
 
-fn run_internal<F>(
-    code: &str,
-    expr: &str,
-    event_cb: F,
-    shots: u32,
-) -> Result<(), Box<stateful::Error>>
+fn run_internal<F>(sources: SourceMap, event_cb: F, shots: u32) -> Result<(), Box<stateful::Error>>
 where
     F: FnMut(&str),
 {
-    let source_name = "code";
+    let source_name = sources
+        .find_by_offset(0)
+        .expect("There must be a source to process")
+        .name
+        .to_string();
```

Comment:
`find_by_offset(0)` returns the entry expression if there is one, which is probably not want you want. You'll want to get the name of the first _file_, from the original array